website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+                m('p', m('.btn.btn-default', {
+                    onclick : self.updateFilter.bind(null, self.systemCollections[0])
+                },' Reload \'All My Projects\''))
+            ]));
+            self.data().data = [];
+            self.refreshView(false);
+            throw new Error('Receiving initial data for File Browser failed. Please check your url');
+        };
+        self.generateFiltersList = function _generateFilterList () {
+            self.users = {};
+            self.tags = {};
+            self.data().data.map(function _generateFiltersListMap(item){
+                var contributors = item.embeds.contributors.data ? item.embeds.contributors.data : [];
+                for(var i = 0; i < contributors.length; i++) {
+                    var u = contributors[i];
+                    if(self.users[u.id] === undefined) {
+                        self.users[u.id] = {
+                            data : u,
+                            count: 1
+                        };
+                    } else {
+                        self.users[u.id].count++;
+                    }
+                }
+
+                var tags = item.attributes.tags ? item.attributes.tags : [];
+                for(var j = 0; j < tags.length; j++) {
+                    var t = tags[j];
+                    if(self.tags[t] === undefined) {
+                        self.tags[t] = 1;
+                    } else {
+                        self.tags[t]++;
+                    }
+                }
+            });
+
+            // Add to lists with numbers
+            for (var user in self.users){
+                var u2 = self.users[user];
+                self.nameFilters.push(new LinkObject('name', { id : u2.data.id, count : u2.count, query : { 'related_counts' : true }}, u2.data.embeds.users.data.attributes.full_name));
+            }
+            for (var tag in self.tags){
+                var t2 = self.tags[tag];
+                self.tagFilters.push(new LinkObject('tag', { tag : tag, count : t2, query : { 'related_counts' : true }}, tag));
+            }
+        };
+
+        // BREADCRUMBS
+        self.updateBreadcrumbs = function _updateBreadcrumbs (linkObject){
+            if (linkObject.type === 'collection' || linkObject.type === 'name' || linkObject.type === 'tag'){
+                self.breadcrumbs([linkObject]);
+                return;
+            }
+            if (linkObject.placement === 'breadcrumb'){
+                self.breadcrumbs().splice(linkObject.index+1, self.breadcrumbs().length-linkObject.index-1);
+                return;
+            }
+            if(linkObject.ancestors.length > 0){
+                linkObject.ancestors.forEach(function(item){
+                    var ancestorLink = new LinkObject('node', item.data, item.data.name);
+                    self.breadcrumbs().push(ancestorLink);
+                });
+            }
+            self.breadcrumbs().push(linkObject);
+        }.bind(self);
```

Comment:
Minor: the `bind` probably isn't necessary, since you use `self` in the method anyway.